src/kem/kyber768/ind_cpa.rs

```diff
@@ -194,11 +198,10 @@ pub(crate) fn encrypt(
     randomness: &[u8; 32],
 ) -> Result<CiphertextCpa, BadRejectionSamplingRandomnessError> {
     // tˆ := Decode_12(pk)
-    let mut t_as_ntt_ring_element_bytes = public_key.chunks(BITS_PER_RING_ELEMENT / 8);
+    let mut t_as_ntt_ring_element_bytes = public_key.chunks(BYTES_PER_RING_ELEMENT);
     let mut t_as_ntt = [KyberPolynomialRingElement::ZERO; RANK];
     for i in 0..t_as_ntt.len() {
-        t_as_ntt[i] = deserialize_little_endian(
-            12,
+        t_as_ntt[i] = deserialize_little_endian_12(
             t_as_ntt_ring_element_bytes.next().expect(
```

Comment:
Couldn't we use the iterator on the `public_key` here as well?

Reply:
Yup, changed it in `ind_cpa.rs` whereever I could.